block-producer/src/state_view/mod.rs

```diff
@@ -186,11 +188,10 @@ fn ensure_tx_inputs_constraints(
                 });
             }
         },
+        // Having `None` here signifies that we are operating
+        // on the state of a new account
```

Comment:
Similarly, here I would change the comment slightly to maybe something like this: "if the account is not in the store, it must be a new account".